### DISCUSSION TITLE
I2S -> MP3 -> icecast server or streammyaudio.com

### DISCUSSION THREAD

**Discussion by z-l-p** [pschatzmann/arduino-audio-tools#674] [Ideas]
First, many thanks for this ever-expanding library. So far I've built an ESP32 webradio receiver that I use literally every day!
I have been working on streaming remote audio from places with unreliable power. I usually use Raspberry Pis with darkIce to connect to icecast servers, but it would be amazing to use an ESP32 instead (less power, more reliable, no risk of SD card corruption...) My idea/question has 2 parts:

- Can we stream MP3-encoded audio to an icecast server? Sort of like [streams-audiokit-webserver_mp3](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-audiokit/streams-audiokit-webserver_mp3) but with an icecast destination? (sorry if this is already implemented somewhere)
- I often work with students who want to stream audio publicly without setting up an icecast server. There is a brilliant project called [StreamMyAudio](https://github.com/schollz/streammyaudio) that uses the author's free public server to forward compressed audio to any number of clients. It's intended to work with ffmpeg piped to curl,  as seen here: 

```
ffmpeg -f alsa -i hw:0 -f mp3 - | \
    curl -s -k -H "Transfer-Encoding: chunked" -X POST -T - \
    "https://streammyaudio.com/YOURSTATIONNAME.mp3?stream=true&advertise=true"

```
It seems like the building blocks are there to reproduce this using your library (especially AudioServer), but I admit that I don't understand enough to make it happen myself.

**Comment by pschatzmann** [comment]
The PI is much more powerfull and has more memory, so it is better suited for streaming. 
Here are the scenarios that are currently supported by my library on a ESP32:
- Built in Server that "streams" audio to one receiver
- If you replace the build-in server with my [TinyHttp library](https://github.com/pschatzmann/TinyHttp) you can stream to multiple clients. (See [examples](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-webserver) that end with Ex)
- Here are some examples which are streaming via [rtsp](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-communication/rtsp). The sample rate however is limited!
- I am providing an [mp3 encoder](https://github.com/pschatzmann/arduino-liblame), but you will most likely run into memory issues. For details see the readme.
